### PR TITLE
gateway: set up fullpath log format

### PR DESCRIPTION
**Pull request by cnvergence**
fixes: https://github.com/envoyproxy/gateway/issues/1216

### CODE REVIEW

**Comment by arkodg** [comment]
> hey @cnvergence can you post what the logs looked like before and after, TIA !

(they should probably be available in CI logs)

**Comment by cnvergence** [comment]
Sure, @arkodg 
Logs were looking like this before:
```bash
envoy-gateway-6df79575c5-4f7cd envoy-gateway 2023-03-31T18:03:56.308Z	INFO	controller/controller.go:228	Starting workers	{"runner": "provider", "controller": "gatewayapi", "worker count": 1}
envoy-gateway-6df79575c5-4f7cd envoy-gateway 2023-03-31T18:03:56.559Z	INFO	runner/runner.go:51	started	{"runner": "infrastructure"}
envoy-gateway-6df79575c5-4f7cd envoy-gateway 2023-03-31T18:03:56.560Z	INFO	runner/runner.go:183	loaded TLS certificate and key	{"runner": "xds-server"}
envoy-gateway-6df79575c5-4f7cd envoy-gateway 2023-03-31T18:03:56.560Z	INFO	runner/runner.go:88	started	{"runner": "xds-server"}
```
With FullCallerEncoder logs are looking like this:
```bash
envoy-gateway-6df79575c5-zrp8m envoy-gateway 2023-03-31T17:59:33.410Z	INFO	/Users/user/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/internal/controller/controller.go:228	Starting workers	{"runner": "provider", "controller": "gatewayapi", "worker count": 1}
envoy-gateway-6df79575c5-zrp8m envoy-gateway 2023-03-31T17:59:33.661Z	INFO	/Users/user/go/src/github.com/envoyproxy/gateway/internal/infrastructure/runner/runner.go:51	started	{"runner": "infrastructure"}
envoy-gateway-6df79575c5-zrp8m envoy-gateway 2023-03-31T17:59:33.661Z	INFO	/Users/user/go/src/github.com/envoyproxy/gateway/internal/xds/server/runner/runner.go:183	loaded TLS certificate and key{"runner": "xds-server"}
envoy-gateway-6df79575c5-zrp8m envoy-gateway 2023-03-31T17:59:33.661Z	INFO	/Users/user/go/src/github.com/envoyproxy/gateway/internal/xds/server/runner/runner.go:88	started	{"runner": "xds-server"}
```

File: internal/log/log.go [hunk 1]
```diff
@@ -9,12 +9,36 @@ import (
...
+			StacktraceKey:  "S",
+			LineEnding:     zapcore.DefaultLineEnding,
+			EncodeLevel:    zapcore.CapitalLevelEncoder,
+			EncodeTime:     zapcore.ISO8601TimeEncoder,
+			EncodeDuration: zapcore.StringDurationEncoder,
+			EncodeCaller:   zapcore.FullCallerEncoder,
+		},
+		OutputPaths:      []string{"stderr"},
+		ErrorOutputPaths: []string{"stderr"},
+	}
```
Comment (zirain) [inline internal/log/log.go:36]:
```suggestion
	zapConfig := zap.NewDevelopmentConfig()
        zapConfig.EncoderConfig.EncodeCaller = zapcore.FullCallerEncoder
```

**Review by zirain — COMMENTED** [review]
No strong option, should make this with a feature gate?

**Comment by cnvergence** [comment]
> No strong option, should make this with a feature gate?

That's a good idea, if the long paths will make it harder to read through logs.

**Comment by misstick86** [comment]
@arkodg sorry, I don't describe this issue clearly.  I also don't think setup fullpath is good idea. like i said the original issue. it's too long and not suitable to read. if we want set relative path may be it's necessary to  customize `zap caller`. or we can open this issue and wait for `zap` package support relative path setting.  somebody has other idea just comment on below.

**Review by kflynn — CHANGES_REQUESTED** [review]
This should be configurable, not forced for everyone, I think.

**Comment by kflynn** [comment]
Closing since we're also closing #1216.

**Pull request closed without merging**